Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -47,180 +47,175 @@ COREGRAPHICS_EXPORT void _CGImageAddDestructionListener(CGImageDestructionListen
+struct __CGImage : CoreFoundation::CppBase<__CGImage> {
+    ComPtr<IWICBitmap> _bitmapImageSource;
+
+    // Calculated from pixel format + WIC backing
+    WICPixelFormatGUID _pixelFormat;
+    woc::StrongCF<CGColorSpaceRef> _colorSpace;
+    size_t _width;
+    size_t _height;
+    size_t _bitsPerPixel;
+    size_t _bitsPerComponent;
+    size_t _bytesPerRow;
+    CGBitmapInfo _bitmapInfo;
+    CGImageAlphaInfo _alphaInfo;
+
+    // User-specified
+    bool _isMask;
+    bool _interpolate;
+    CGColorRenderingIntent _renderingIntent;
+
+    __CGImage()
+        : Parent(),
+          _bitmapImageSource(nullptr),
+          _pixelFormat(GUID_WICPixelFormatUndefined),
+          _width(0),
+          _height(0),
+          _bitsPerPixel(0),
+          _bitsPerComponent(0),
+          _bytesPerRow(0),
+          _bitmapInfo(kCGBitmapByteOrderDefault),
+          _alphaInfo(kCGImageAlphaNone),
+          _isMask(false),
+          _interpolate(false),
+          _renderingIntent(kCGRenderingIntentDefault) {
+    }
+
+    // We would use a copy constructor here, but they cannot return HRESULTs.
+    // There are no other throwing constructors in our code, so it doesn't make sense to introduce our first one.
+    HRESULT CopyFromImage(const __CGImage& other, WICPixelFormatGUID newFormat = GUID_WICPixelFormatDontCare, CGRect* region = nullptr) {
+        CGRect rect = region ? *region : CGRect{ CGPointZero, { other._width, other._height } };
+
+        ComPtr<IWICImagingFactory> imageFactory;
+        RETURN_IF_FAILED(_CGGetWICFactory(&imageFactory));
+
+        ComPtr<IWICBitmapSource> bitmapSource;
+
+        if (newFormat == GUID_WICPixelFormatDontCare || newFormat == other._pixelFormat) {
+            bitmapSource = other.ImageSource();
+        } else {
+            ComPtr<IWICFormatConverter> converter;
+            RETURN_IF_FAILED(imageFactory->CreateFormatConverter(&converter));
+
+            RETURN_IF_FAILED(converter->Initialize(
+                other.ImageSource(), newFormat, WICBitmapDitherTypeNone, nullptr, 0.f, WICBitmapPaletteTypeMedianCut));
+
+            bitmapSource = converter;
+        }
 
-    inline WICPixelFormatGUID PixelFormat() const {
-        WICPixelFormatGUID pixelFormat;
-        RETURN_RESULT_IF_FAILED(bitmapImageSource->GetPixelFormat(&pixelFormat), GUID_WICPixelFormatUndefined);
-        return pixelFormat;
-    }
+        ComPtr<IWICBitmap> newBitmap;
+        RETURN_IF_FAILED(imageFactory->CreateBitmapFromSourceRect(bitmapSource.Get(), rect.origin.x, rect.origin.y, rect.size.width, rect.size.height, &newBitmap));
 
-    inline const __CGImagePixelProperties* Properties() const {
-        WICPixelFormatGUID pixelFormat = PixelFormat();
-        return _CGGetPixelFormatProperties(pixelFormat);
-    }
+        RETURN_IF_FAILED(SetImageSource(newBitmap.Get())); // also sets _colorSpace
+        _isMask = other._isMask;
+        _interpolate = other._interpolate;
+        _renderingIntent = other._renderingIntent;
```

Comment:
hmmm i still feel reluctant about this, could we get a method like copy properties so everyone can use that.
This way a future developer who tries to manually copy over properties will not miss it.